sentry-core/src/client.rs

```diff
@@ -239,14 +240,33 @@ impl Client {
             if self.sample_should_send() {
                 if let Some(event) = self.prepare_event(event, scope) {
                     let event_id = event.event_id;
-                    transport.send_envelope(event.into());
+                    let session = scope.and_then(|scope| {
+                        if let SessionUpdate::NeedsFlushing(session) =
+                            scope.update_session_from_event(&event)
+                        {
+                            Some(session)
+                        } else {
+                            None
+                        }
+                    });
+                    let mut envelope: Envelope = event.into();
```

Comment:
If you move this to before the `let session` you don't need the if statement on the next line and can and it right away when you get `NeedsFlushing` (and don't need the else branch there either)

Reply:
`Scope::update_session_from_event` needs a `&Event`, but `event.into()` consumes it. The second `if let` is unfortunate, but we do need it because of lifetimes.